pkgs/by-name/ya/yazi/package.nix

```diff
@@ -56,30 +57,34 @@ let
       null
     else
       runCommand "YAZI_CONFIG_HOME" { } ''
-        mkdir -p $out
+        mkdir -p "$out"
         ${lib.concatMapStringsSep "\n" (
           name:
           lib.optionalString (settings ? ${name} && settings.${name} != { }) ''
-            ln -s ${settingsFormat.generate "${name}.toml" settings.${name}} $out/${name}.toml
+            ln -s ${settingsFormat.generate "${name}.toml" settings.${name}} "$out/${name}.toml"
           ''
         ) files}
 
-        mkdir $out/plugins
+        mkdir "$out/plugins"
         ${lib.optionalString (plugins != { }) ''
           ${lib.concatStringsSep "\n" (
-            lib.mapAttrsToList (name: value: "ln -s ${value} $out/plugins/${name}") plugins
+            lib.mapAttrsToList (name: value: ''
+              mkdir -p "$out/plugins/${name}"
+              ln -s ${value}/main.lua "$out/plugins/${name}/main.lua"
+            '') plugins
           )}
+          ${lib.optionalString (mainLua != null) "ln -s ${mainLua} \"$out/plugins/main.lua\""}
```

Comment:
I think there's still a misunderstanding. There plugins directory just contains the plugins and they have the `main.lua` entrypoint. There isn't a `main.lua` in the plugins directory. https://yazi-rs.github.io/docs/plugins/overview